app/classifier.py

```diff
@@ -1,18 +1,56 @@
+from operator import itemgetter
+import logging
+import string
+import time
+import os
+
+import numpy as np
+import tensorflow as tf
+
 from app import embedder
+from app import model_fetcher
+
+
+def fetchVocab(vocab: str):
+    with open(vocab, 'r') as f:
+        return f.readlines()
 
 
 class Classifier:
 
-    def __init__(self):
-        self.embedder = embedder.Embedder()
+    def __init__(self, bert, classifier, vocab: str, prefetch=False):
+        self.logger = logging.getLogger('app.logger')
+        if prefetch:
+            f = model_fetcher.Fetcher()
+            filenames = f.fetchAll()
+            for f in filenames:
+                self.logger.info(f)
+        self.vocab = fetchVocab(vocab)
+        self.classifier = classifier
+        self.embedder = embedder.Embedder(bert)
+        self.predictor = tf.contrib.predictor.from_saved_model(self.classifier)
+
+    def classify(self, seq: str):
+        embedding = self.embedder.GetEmbedding(seq)
+        input_mask = [1] * embedding.shape[0]
 
-    def classify(self):
-        # calculate UID of seqs
-        # fetch embedding matrix from the cache
-        #   if it exists, return it
-        #   create and store embedding, return it
         # classify seq, with its embedding matrix, using a specific model
-        # filter results
+        features = {
+            "embeddings": np.expand_dims(embedding, axis=0),
+            "input_mask": np.expand_dims(input_mask, axis=0),
+        }
+
+        predictions = self.predictor(features)
+        results = predictions["probabilities"][0]
+        self.logger.debug(results)
+
         # map results back to topic strings, according to classifier metadata
-        # return topic confidence array
-        return
+        out = list(zip((t.rstrip() for t in self.vocab), results))
```

Comment:
I can't read this line ;)

Reply:
I've tried to make it clearer with a more exemplary comment. LMK.